include/nfc/tnep/tnep.rst

```diff
@@ -0,0 +1,123 @@
+.. _tnep_tag_readme:
+
+Tag NDEF Exchange Protocol for NFC Tag Device
+#############################################
+
+TNEP (The Tag NDEF Exchange Protocol) is an application-level protocol for sending or
+retrieving application data units, in the form of NFC Data Exchange Format (NDEF) messages,
+between one Reader/Writer and an NFC Tag Device. It operates between Type X Tag and NDEF application layer.
+
+Initialization
+==============
+Before using the TNEP library in a tag device application it is required to set the NDEF message buffers and define TNEP services.
+
+NDEF Message
+------------
+The TNEP library communicates with Type X Tag using NDEF message buffers.
+The buffers are initialized using :cpp:func:`nfc_tnep_tag_tx_msg_buffer_register`.
+It is required to provide a pointer to the two buffers and a maximum buffer size in bytes.
+Linking the buffers with Type X Tag layer should be done by a user application. This library
+needs two TX buffer because when one buffer is used to emulate Tag NDEF Message, in second
+the new NDEF Message is encoded.
+
+When the NFC Polling Device write a new NDEF Message inside the Tag buffer the application should inform the TNEP library
+calling :cpp:func:`nfc_tnep_tag_rx_msg_indicate`.
+
+Services
+--------
+The word services mean the capabilities and features provided to the adjacent upper layer.
```

Comment:
What are these capabilities and features in the case of NFC TNEP Tag library? Can you name some examples?